docs/DeveloperGuide.md

```diff
@@ -1,34 +1,600 @@
+The command: 
+
+    add appt /ic S7777777Z /d 01/04/2020 /t 12:00:00 /r Checkup 
+    
+will add an appointment with `NRIC` as S7777777Z with the following attributes: 
+* appointment date: `01/04/2020`
+* appointment time: `12:00:00`
+* reason for appointment: `Checkup`
+
+An `appointmentId` will also be given when an appointment is successfully added. 
+
+**Implementation** 
+
+![Add Appointment Sequence Diagram](images/AddAppointmentSequenceDiagram.jpg)
+
+The following steps explains the sequence of events: 
+
+1. The user enters `add appt /ic S7777777Z /d 01/04/2020 /t 12:00:00 /r Checkup` .
+
+2. `HappyPills` calls `Parser#parse()` which then calls `AppointmentParser#parse()`. 
+
+3. `AppointmentParser#parse()` will create an object `AddAppointmentCommand`.
+
+5. `HappyPills` then calls `AddAppointmentCommand.execute()` to execute the method. 
+
+6. In `AddAppointmentCommand.execute()`, an appointment is added and the display message is returned.
+
+#### 4.3.2. Edit Appointment 
+
+#### 4.3.3. Delete Appointment 
+
+The user can delete an appointment from the list of appointments currently in the program. The command: 
+
+    delete appt 1 
+    
+will delete the appointment with appointment ID `1`, if found. 
+
+**Implementation** 
+
+The following steps explains the sequence of events: 
+
+1. The user enters `delete appt 1`. 
+
+2. `HappyPills` calls `Parser#parse()` which then calls `AppointmentParser#parse()`.
+
+3. `AppointmentParser#parse()` will then create an object `DeleteAppointmentCommand`. 
+
+4. `HappyPills` then calls `DeleteAppointmentCommand.execute()` to execute the command. 
+
+5. In `DeleteAppointmentCommand.execute()`, the respective patient is deleted and the display message is returned. 
+
+#### 4.3.4. Mark Appointment as Done 
+
+#### 4.3.5. List Appointments 
+
+The user is able to get a list of all the appointments currently in the program.
+The command: 
+    
+    list appt
+    
+will list all the appointments in the `AppointmentMap`. 
+
+**Implementation** 
+
+![List Appointment Sequence Diagram](images/ListAppointmentSequenceDiagram.jpg)
+
+The following steps explains the sequence of events: 
+
+1. The user enters `list appt`.
+
+2. `HappyPills` calls `Parser#parse()` which then calls `AppointmentParser#parse()`.
+
+3. `AppointmentParser#parse()` will create an object `ListAppointmentCommand`.
+
+4. `HappyPills` then calls `ListAppointmentCommand.execute()` to execute the method. 
+
+5. In `ListAppointmentCommand.execute()`, it calls `TextUi.getAppointmentList()` which is then returned as the display message. 
+
+#### 4.3.6. Find Appointments of Patient
+
+The user is able to find all the appointments a specific patient has.
+The command: 
+    
+    find appt S7777777Z
+    
+will list all the appointments that the patient with NRIC S7777777Z has. 
+
+**Implementation**
+
+![Find Appointment Sequence Diagram](images/FindAppointmentSequenceDiagram.jpg)
```

Comment:
Like in `images/AddPatientSequenceDiagram.jpg`, I suggest shortening the activation bars of the methods, as a method cannot remain active after it has returned.

From the CS2113/T website:
![bug1](https://user-images.githubusercontent.com/35903974/77999573-77fa3980-7365-11ea-8274-f6f23cffa7b2.png)